samza-kv/src/test/scala/org/apache/samza/storage/kv/TestKeyValueStorageEngine.scala

```diff
@@ -142,12 +142,22 @@ class TestKeyValueStorageEngine {
   @Test
   def testRestoreMetrics(): Unit = {
```

Comment:
maybe add a test case to ensure the transition from `TRIM` to `RESTORE` doesn't happen.

Reply:
Will add in PR 5.